hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -689,15 +671,24 @@ def _validate_phases(phases):
 
 settings.define_setting(
     'perform_health_check',
-    default=True,
+    default=not_set,
     description=u"""
 If set to True, Hypothesis will run a preliminary health check before
 attempting to actually execute your test.
-"""
+""",
+    deprecation_message="""
+This setting is deprecated, as `perform_health_check=False` duplicates the
+effect of `suppress_health_check=list(HealthCheck)`.  Use that, or a more
+specific list, instead!
+""",
 )
 
 
 def validate_health_check_suppressions(suppressions):
+    if suppressions == list(HealthCheck):
```

Comment:
I feel like there should be a better way to do this. Ideally I'd like it so that `list(HealthCheck)` just doesn't include these, which is perfectly possible but requires a gross amount of work.

Reply:
(horrified fascination)  Tell me more?

Reply:
> (horrified fascination)

Oh, you just extend `enum.EnumMeta` and override the `__iter__` method to call super and filter out the undesired suppressions, then use that as the metaclass for `HealthCheck` (which requires some annoying compatibility shenanigans to get right because the syntax for that changed in Python 2 and 3).

...

Look I *said* it was gross.

Reply:
Oh of *course*, why didn't I think of messing with the metaclass?  (only half sarcastic!)

Personally I find the current workaround less offensive, especially as we're going to remove those old options sometime soon... but if you really want it I'll do the gross thing :stuck_out_tongue_closed_eyes:

Reply:
> Personally I find the current workaround less offensive, given that we're going to remove those old options sometime soon - simply less churn to deal with.

Yeah, I don't actually like the idea of messing with the metaclass either.

That being said, here's one thing I don't like about this solution: I think it's pretty natural to want to white list which health checks you *want* to include, and the obvious way to write that is something like `suppress_health_check=set(HealthCheck) - my_white_list`. People who want to do this can now only avoid the deprecation warning by writing code that will break when the deprecated functionality goes away!

Maybe we should add a classmethod to `HealthCheck` that returns all non-deprecated health checks and recommend that instead of `list(HealthCheck)`?

Reply:
> Maybe we should add a classmethod to HealthCheck that returns all non-deprecated health checks and recommend that instead of list(HealthCheck)?

This sounds like the best option to me - clean now, easy to support in future (when removing deprecated checks or deprecating more), and with an absolute minimum of magic.